providers/async_gemini.py

```
@@ -0,0 +1,610 @@
+                # This is a fallback for when called from a running event loop.
+                future = asyncio.run_coroutine_threadsafe(self.count_tokens_async(text, model_name), loop)
+                return future.result(timeout=30)  # 30 second timeout
+            else:
+                return asyncio.run(self.count_tokens_async(text, model_name))
+        except RuntimeError:
+            # No event loop running, create a new one
+            return asyncio.run(self.count_tokens_async(text, model_name))
+        except Exception as e:
+            logger.warning(f"Failed to count tokens asynchronously: {e}")
+            # Fall back to simple estimation
+            resolved_name = self._resolve_model_name(model_name)
```

Flake8 (7.2.0)

[error] local variable 'resolved_name' is assigned to but never used

(F841)